clash-cores/src/Clash/Cores/Sgmii/Common.hs

```diff
@@ -0,0 +1,140 @@
+-- |
+--   Copyright   :  (C) 2024, QBayLogic B.V.
+--   License     :  BSD2 (see the file LICENSE)
+--   Maintainer  :  QBayLogic B.V. <devops@qbaylogic.com>
+--
+--   Common functions, type definitions and hard-coded settings used in the
+--   different modules that are defined for SGMII
+module Clash.Cores.Sgmii.Common where
+
+import Clash.Cores.LineCoding8b10b
+import Clash.Prelude
+
+-- | Format of a single code group, 10-bit
+type Cg = BitVector 10
+
+-- | Format of @rxConfReg@ and @txConfReg@, size of two data words
+type ConfReg = BitVector 16
+
+-- | Defines the type of the signal that indicates whether the current received
+--   code group is at an even or odd index in the sequence
+data Even = Even | Odd
+  deriving (Generic, NFDataX, Eq, Show)
+
+-- | Function that makes an Even RxEven Odd, and vice-versa
+nextEven :: Even -> Even
+nextEven Even = Odd
+nextEven Odd = Even
+
+-- | Link speed that was communicated by the PHY
+data LinkSpeed = Speed10 | Speed100 | Speed1000
+  deriving (Generic, NFDataX, BitPack)
+
+-- | Get the current link speed from a 'ConfReg'
+toLinkSpeed :: ConfReg -> LinkSpeed
+toLinkSpeed confReg =
+  unpack $ pack (testBit confReg 11) ++# pack (testBit confReg 10)
+
+-- | Defines the possible different types of ordered sets that can be generated
+--   by the 'Sgmii.PcsTransmit.orderedSet' process
+data OrderedSet = OSetC | OSetI | OSetR | OSetS | OSetT | OSetV | OSetD
+  deriving (Generic, NFDataX, Eq, Show)
+
+-- | Defines the possible values for the RUDI output signal of the PCS Receive
+--   block as defined in IEEE 802.3 Clause 36
+data Rudi = C ConfReg | I | Invalid
```

Comment:
Could `I` be renamed to `I_recv`  or something like it? When I was reading it in other places I interpreted it as `1`.

Reply:
I have prefixed all of them with `Rudi` now.